samples/cellular/modem_shell/src/sock/sock_shell.c

```diff
@@ -165,13 +172,77 @@ static const char sock_rai_usage_str[] =
 	"                            option before the next send call.\n"
 	"  -h, --help,               Shows this help information";
 
+static const char sock_setopt_usage_str[] =
+	"Usage: sock option set -i <socket id> -o <socket option> -v <value>\n"
+	"       [--level <level>]\n"
+	"Options:\n"
+	"  -i, --id, [int]           Socket id. Use 'sock list' command to see open\n"
+	"                            sockets.\n"
+	"  -o, --option, [str|int]   String name or integer ID of the socket option\n"
+	"      --level, [str|int]    String name or integer ID of the socket option level.\n"
+	"                            This is optional. If a valid option string is given\n"
+	"                            with '-o' option, the level is set automatically.\n"
+	"  -v, --value, [str|int]    Value of the socket option\n"
+	"  -h, --help,               Shows this help information\n";
+
+static const char sock_getopt_usage_str[] =
+	"Usage: sock option get -i <socket id> -o <socket option> [--level <level>]\n"
+	"Options:\n"
+	"  -i, --id, [int]           Socket id. Use 'sock list' command to see open\n"
+	"                            sockets.\n"
+	"  -o, --option, [str|int]   String name or integer ID of the socket option\n"
+	"      --level, [str|int]    String name or integer ID of the socket option level.\n"
+	"                            This is optional. If a valid option string is given\n"
+	"                            with '-o' option, the level is set automatically.\n"
+	"  -h, --help,               Shows this help information\n";
+
+static const char sock_option_id_list_str[] =
+	"List of supported socket options which you can set as string for '-o' option:\n"
+	" - TLS_SEC_TAG_LIST, [str] (for example \"42,43,44\")\n"
+	" - TLS_HOSTNAME, [str]\n"
+	" - TLS_CIPHERSUITE_LIST, [str] (for example \"0xC024,0xC023\")\n"
+	" - TLS_PEER_VERIFY, [int]\n"
+	" - TLS_DTLS_ROLE, [int]\n"
+	" - TLS_SESSION_CACHE, [int]\n"
+	" - TLS_SESSION_CACHE_PURGE, (set only)\n"
```

Comment:
Should this list be splitted for setopt and getopt to avoid listing options not relevant for the used option command?

Reply:
I agree that it would be nicer, but it would also consume something like 1 kB of more flash. @trantanen, opinions?

Reply:
I wouldn't do it due to the memory it takes.

Reply:
I tend to agree, so lets keep it as it is.

Reply:
Could be made with a list of options having get/set flags and looping this list when printing. But ok, that is probably overkill here.